byterun/caml/sys.h

```diff
@@ -26,8 +26,14 @@ extern "C" {
 
 #define NO_ARG Val_int(0)
 
-CAMLextern void caml_sys_error (value);
-CAMLextern void caml_sys_io_error (value);
+CAMLnoreturn_start
+CAMLextern void caml_sys_error (value)
+CAMLnoreturn_end;
+
+CAMLnoreturn_start
+CAMLextern void caml_sys_io_error (value)
+CAMLnoreturn_end;
+
 CAMLextern double caml_sys_time_unboxed(value);
 CAMLextern void caml_sys_init (char_os * exe_name, char_os ** argv);
 CAMLextern value caml_sys_exit (value);
```

Comment:
Should `caml_sys_exit` be considered noreturn too?

Reply:
It should, except that if the "C plugins" mechanism is active (`CAML_WITH_CPLUGINS` is defined), there is no guarantee that `caml_sys_exit` will not return.

We could add more #ifdefs to declare `caml_sys_exit` specially if `CAML_WITH_CPLUGINS` is not defined.  We could also get rid of this "C plugins" instrumentation hell.

Reply:
So the warnings from the static analyser will still strike if we supply an OCaml build with `CAML_WITH_CPLUGINS` defined :-/  We're very likely to be compiling MirageOS with the plugins option disabled soon since we trust it about as far as we can throw a small stick.

Reply:
`CAML_WITH_CPLUGINS` is no longer defined by default.  Since 4.06, you need to `configure -with-cplugins` to activate it.  Still, at some future point we should review the actual uses of this mechanism and decide whether to keep it or not.